common/archiver/archivalMetadata.go

```diff
@@ -154,7 +210,7 @@ func (a *archivalConfig) GetDomainDefaultURI() string {
 
 func (a *archivalConfig) isValid() bool {
 	URISet := len(a.domainDefaultURI) != 0
-	return (URISet && a.ClusterConfiguredForArchival()) || (!URISet && !a.ClusterConfiguredForArchival())
+	return (URISet && a.ClusterConfiguredForArchival()) || (!a.enableRead && a.domainDefaultStatus == shared.ArchivalStatusDisabled && !URISet && !a.ClusterConfiguredForArchival())
```

Comment:
Can you factor this out into three lines

validEnabled := ...
validDisabled := ...
return validEnabled || validDisabled